adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -539,6 +639,70 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
   }
 
   /**
+   * Prepares two RDDs to be joined with any shuffleRegionJoin. This includes copartition
+   * and sort of the rightRdd if necessary.
+   *
+   *  @param genomicRdd The RDD to join to.
+   *  @return A tuple containing both RDDs ready to join, the partitionMap and the number
+   *          of partitions.
+   *          _._1 is the leftRdd,
+   *          _._2 is the rightRdd,
+   *          _._3 is the partitionMap for the joined RDDs,
+   *          _._4 is the number of partitions
+   */
+  private def prepareForShuffleRegionJoin[X, Y <: GenomicRDD[X, Y], Z <: GenomicRDD[(T, X), Z]](
+    genomicRdd: GenomicRDD[X, Y],
+    optPartitions: Option[Int] = None)(
+      implicit tTag: ClassTag[T], xTag: ClassTag[X]): (RDD[(ReferenceRegion, T)], RDD[(ReferenceRegion, X)], Option[Seq[Option[(ReferenceRegion, ReferenceRegion)]]], Int) = {
+
+    // did the user provide a set partition count?
+    // if no, take the max partition count from our rdds
+    val partitions = optPartitions.getOrElse(this.rdd.partitions.length)
+
+    val leftRdd: GenomicRDD[T, U] = if (sorted) {
+      if (partitions != rdd.partitions.length) {
+        evenlyRepartition(partitions)
+      } else {
+        this
+      }
+    } else {
+      repartitionAndSort(partitions)
```

Comment:
We don't actually have to. I wrote copartitionByReferenceRegion to accept unsorted RDDs so we are only repartitioning the right RDD once instead of sort/repartitioning then copartitioning. 

Line 669 is for the leftRdd, just in case that was the line in question.